### ISSUE TITLE
Completely broken

### ISSUE THREAD

**Issue by atwright147** [HuakunShen/tauri-plugin-system-info#5]
[error.txt](https://github.com/user-attachments/files/15539420/error.txt)
I am no Rust or Tauri dev (just trying to learn at the mo). In a fresh Tauri install I get the following:

<details>
<summary>Error Snippet</summary>
```sh
               i128
             and 261 others
note: required by a bound in `aggregate::_::_serde::ser::SerializeStruct::serialize_field`
    --> /Users/andy/.cargo/registry/src/index.crates.io-6f17d22bba15001f/serde-1.0.203/src/ser/mod.rs:1860:21
     |
1858 |     fn serialize_field<T>(&mut self, key: &'static str, value: &T) -> Result<(), Self::Error>
     |        --------------- required by a bound in this associated function
1859 |     where
1860 |         T: ?Sized + Serialize;
     |                     ^^^^^^^^^ required by this bound in `SerializeStruct::serialize_field`
     = note: this error originates in the derive macro `Serialize` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `uom::si::Quantity<dyn uom::si::Dimension<I = Z0, J = Z0, Kind = (dyn uom::Kind + 'static), L = PInt<UInt<UInt<UTerm, B1>, B0>>, M = PInt<UInt<UTerm, B1>>, N = Z0, T = NInt<UInt<UInt<UTerm, B1>, B1>>, Th = Z0>, dyn uom::si::Units<f32, amount_of_substance = uom::si::amount_of_substance::mole, electric_current = starship_battery::units::electric_current::ampere, length = uom::si::length::meter, luminous_intensity = uom::si::luminous_intensity::candela, mass = uom::si::mass::kilogram, thermodynamic_temperature = starship_battery::units::thermodynamic_temperature::kelvin, time = starship_battery::units::time::second>, f32>: Serialize` is not satisfied
    --> /Users/andy/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tauri-plugin-system-info-0.2.0/src/model/common.rs:313:17
     |
313  | #[derive(Debug, Serialize, Deserialize)]
     |                 ^^^^^^^^^ the trait `Serialize` is not implemented for `Quantity<dyn Dimension<I = Z0, J = Z0, Kind = ..., L = ..., M = ..., N = ..., T = ..., Th = ...>, ..., ...>`
...
319  |     energy_rate: Power,
     |     ----------- required by a bound introduced by this call
     |
     = help: the following other types implement trait `Serialize`:
               bool
               char
               isize
               i8
               i16
               i32
               i64
               i128
             and 261 others
note: required by a bound in `aggregate::_::_serde::ser::SerializeStruct::serialize_field`
    --> /Users/andy/.cargo/registry/src/index.crates.io-6f17d22bba15001f/serde-1.0.203/src/ser/mod.rs:1860:21
     |
1858 |     fn serialize_field<T>(&mut self, key: &'static str, value: &T) -> Result<(), Self::Error>
     |        --------------- required by a bound in this associated function
1859 |     where
1860 |         T: ?Sized + Serialize;
     |                     ^^^^^^^^^ required by this bound in `SerializeStruct::serialize_field`
     = note: this error originates in the derive macro `Serialize` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `uom::si::Quantity<dyn uom::si::Dimension<I = NInt<UInt<UTerm, B1>>, J = Z0, Kind = (dyn uom::Kind + 'static), L = PInt<UInt<UInt<UTerm, B1>, B0>>, M = PInt<UInt<UTerm, B1>>, N = Z0, T = NInt<UInt<UInt<UTerm, B1>, B1>>, Th = Z0>, dyn uom::si::Units<f32, amount_of_substance = uom::si::amount_of_substance::mole, electric_current = starship_battery::units::electric_current::ampere, length = uom::si::length::meter, luminous_intensity = uom::si::luminous_intensity::candela, mass = uom::si::mass::kilogram, thermodynamic_temperature = starship_battery::units::thermodynamic_temperature::kelvin, time = starship_battery::units::time::second>, f32>: Serialize` is not satisfied
    --> /Users/andy/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tauri-plugin-system-info-0.2.0/src/model/common.rs:313:17
     |
313  | #[derive(Debug, Serialize, Deserialize)]
     |                 ^^^^^^^^^ the trait `Serialize` is not implemented for `Quantity<dyn Dimension<I = ..., J = ..., Kind = ..., L = ..., M = ..., N = ..., T = ..., Th = ...>, ..., ...>`
...
320  |     voltage: ElectricPotential,
     |     ------- required by a bound introduced by this call
     |
     = help: the following other types implement trait `Serialize`:
               bool
               char
               isize
               i8
               i16
               i32
               i64
               i128
             and 261 others
note: required by a bound in `aggregate::_::_serde::ser::SerializeStruct::serialize_field`
    --> /Users/andy/.cargo/registry/src/index.crates.io-6f17d22bba15001f/serde-1.0.203/src/ser/mod.rs:1860:21
     |
1858 |     fn serialize_field<T>(&mut self, key: &'static str, value: &T) -> Result<(), Self::Error>
     |        --------------- required by a bound in this associated function
1859 |     where
1860 |         T: ?Sized + Serialize;
     |                     ^^^^^^^^^ required by this bound in `SerializeStruct::serialize_field`
     = note: this error originates in the derive macro `Serialize` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `uom::si::Quantity<dyn uom::si::Dimension<I = Z0, J = Z0, Kind = (dyn uom::si::marker::TemperatureKind + 'static), L = Z0, M = Z0, N = Z0, T = Z0, Th = PInt<UInt<UTerm, B1>>>, dyn uom::si::Units<f32, amount_of_substance = uom::si::amount_of_substance::mole, electric_current = starship_battery::units::electric_current::ampere, length = uom::si::length::meter, luminous_intensity = uom::si::luminous_intensity::candela, mass = uom::si::mass::kilogram, thermodynamic_temperature = starship_battery::units::thermodynamic_temperature::kelvin, time = starship_battery::units::time::second>, f32>: Serialize` is not satisfied
    --> /Users/andy/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tauri-plugin-system-info-0.2.0/src/model/common.rs:313:17
     |
313  | #[derive(Debug, Serialize, Deserialize)]
     |                 ^^^^^^^^^ the trait `Serialize` is not implemented for `Quantity<dyn Dimension<I = Z0, J = Z0, Kind = ..., L = ..., M = ..., N = ..., T = ..., Th = ...>, ..., ...>`, which is required by `std::option::Option<uom::si::Quantity<dyn uom::si::Dimension<I = Z0, J = Z0, Kind = (dyn uom::si::marker::TemperatureKind + 'static), L = Z0, M = Z0, N = Z0, T = Z0, Th = PInt<UInt<UTerm, B1>>>, dyn uom::si::Units<f32, amount_of_substance = uom::si::amount_of_substance::mole, electric_current = starship_battery::units::electric_current::ampere, length = uom::si::length::meter, luminous_intensity = uom::si::luminous_intensity::candela, mass = uom::si::mass::kilogram, thermodynamic_temperature = starship_battery::units::thermodynamic_temperature::kelvin, time = starship_battery::units::time::second>, f32>>: Serialize`
...
324  |     temperature_kelin: Option<ThermodynamicTemperature>,
     |     ----------------- required by a bound introduced by this call
     |
     = help: the following other types implement trait `Serialize`:
               bool
               char
               isize
               i8
               i16
               i32
               i64
               i128
             and 261 others
     = note: required for `Option<Quantity<..., ..., ...>>` to implement `Serialize`
note: required by a bound in `aggregate::_::_serde::ser::SerializeStruct::serialize_field`
    --> /Users/andy/.cargo/registry/src/index.crates.io-6f17d22bba15001f/serde-1.0.203/src/ser/mod.rs:1860:21
     |
1858 |     fn serialize_field<T>(&mut self, key: &'static str, value: &T) -> Result<(), Self::Error>
     |        --------------- required by a bound in this associated function
1859 |     where
1860 |         T: ?Sized + Serialize;
     |                     ^^^^^^^^^ required by this bound in `SerializeStruct::serialize_field`
     = note: the full name for the type has been written to '/Users/andy/Development/Tauri/photo-importer/src-tauri/target/debug/deps/tauri_plugin_system_info-f921e2b9bc7ad894.long-type-2250701583780203999.txt'
     = note: consider using `--verbose` to print the full type name to the console
     = note: this error originates in the derive macro `Serialize` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `uom::si::Quantity<dyn uom::si::Dimension<I = Z0, J = Z0, Kind = (dyn uom::Kind + 'static), L = Z0, M = Z0, N = Z0, T = PInt<UInt<UTerm, B1>>, Th = Z0>, dyn uom::si::Units<f32, amount_of_substance = uom::si::amount_of_substance::mole, electric_current = starship_battery::units::electric_current::ampere, length = uom::si::length::meter, luminous_intensity = uom::si::luminous_intensity::candela, mass = uom::si::mass::kilogram, thermodynamic_temperature = starship_battery::units::thermodynamic_temperature::kelvin, time = starship_battery::units::time::second>, f32>: Serialize` is not satisfied
    --> /Users/andy/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tauri-plugin-system-info-0.2.0/src/model/common.rs:313:17
     |
313  | #[derive(Debug, Serialize, Deserialize)]
     |                 ^^^^^^^^^ the trait `Serialize` is not implemented for `Quantity<dyn Dimension<I = Z0, J = Z0, Kind = dyn Kind, L = Z0, M = Z0, N = Z0, T = ..., Th = ...>, ..., ...>`, which is required by `std::option::Option<uom::si::Quantity<dyn uom::si::Dimension<I = Z0, J = Z0, Kind = (dyn uom::Kind + 'static), L = Z0, M = Z0, N = Z0, T = PInt<UInt<UTerm, B1>>, Th = Z0>, dyn uom::si::Units<f32, amount_of_substance = uom::si::amount_of_substance::mole, electric_current = starship_battery::units::electric_current::ampere, length = uom::si::length::meter, luminous_intensity = uom::si::luminous_intensity::candela, mass = uom::si::mass::kilogram, thermodynamic_temperature = starship_battery::units::thermodynamic_temperature::kelvin, time = starship_battery::units::time::second>, f32>>: Serialize`
...
331  |     time_to_full: Option<Time>,
     |     ------------ required by a bound introduced by this call
     |
     = help: the following other types implement trait `Serialize`:
               bool
               char
               isize
               i8
               i16
               i32
               i64
               i128
             and 261 others
     = note: required for `Option<Quantity<..., ..., ...>>` to implement `Serialize`
note: required by a bound in `aggregate::_::_serde::ser::SerializeStruct::serialize_field`
    --> /Users/andy/.cargo/registry/src/index.crates.io-6f17d22bba15001f/serde-1.0.203/src/ser/mod.rs:1860:21
     |
1858 |     fn serialize_field<T>(&mut self, key: &'static str, value: &T) -> Result<(), Self::Error>
     |        --------------- required by a bound in this associated function
1859 |     where
1860 |         T: ?Sized + Serialize;
     |                     ^^^^^^^^^ required by this bound in `SerializeStruct::serialize_field`
     = note: the full name for the type has been written to '/Users/andy/Development/Tauri/photo-importer/src-tauri/target/debug/deps/tauri_plugin_system_info-f921e2b9bc7ad894.long-type-5172835510086939520.txt'
     = note: consider using `--verbose` to print the full type name to the console
     = note: this error originates in the derive macro `Serialize` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `uom::si::Quantity<(dyn uom::si::Dimension<I = Z0, J = Z0, Kind = (dyn uom::Kind + 'static), L = Z0, M = Z0, N = Z0, T = Z0, Th = Z0> + 'static), (dyn uom::si::Units<f32, amount_of_substance = uom::si::amount_of_substance::mole, electric_current = starship_battery::units::electric_current::ampere, length = uom::si::length::meter, luminous_intensity = uom::si::luminous_intensity::candela, mass = uom::si::mass::kilogram, thermodynamic_temperature = starship_battery::units::thermodynamic_temperature::kelvin, time = starship_battery::units::time::second> + 'static), f32>: Deserialize<'_>` is not satisfied
    --> /Users/andy/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tauri-plugin-system-info-0.2.0/src/model/common.rs:315:22
     |
315  |     state_of_charge: Ratio,
     |                      ^^^^^ the trait `Deserialize<'_>` is not implemented for `Quantity<dyn Dimension<I = Z0, J = Z0, Kind = dyn Kind, L = Z0, M = Z0, N = Z0, T = Z0, Th = Z0>, ..., ...>`
     |
     = help: the following other types implement trait `Deserialize<'de>`:
               <bool as Deserialize<'de>>
               <char as Deserialize<'de>>
               <isize as Deserialize<'de>>
               <i8 as Deserialize<'de>>
               <i16 as Deserialize<'de>>
               <i32 as Deserialize<'de>>
               <i64 as Deserialize<'de>>
               <i128 as Deserialize<'de>>
             and 295 others
note: required by a bound in `next_element`
    --> /Users/andy/.cargo/registry/src/index.crates.io-6f17d22bba15001f/serde-1.0.203/src/de/mod.rs:1726:12
     |
1724 |     fn next_element<T>(&mut self) -> Result<Option<T>, Self::Error>
     |        ------------ required by a bound in this associated function
1725 |     where
1726 |         T: Deserialize<'de>,
     |            ^^^^^^^^^^^^^^^^ required by this bound in `SeqAccess::next_element`

error[E0277]: the trait bound `uom::si::Quantity<(dyn uom::si::Dimension<I = Z0, J = Z0, Kind = (dyn uom::Kind + 'static), L = PInt<UInt<UInt<UTerm, B1>, B0>>, M = PInt<UInt<UTerm, B1>>, N = Z0, T = NInt<UInt<UInt<UTerm, B1>, B0>>, Th = Z0> + 'static), (dyn uom::si::Units<f32, amount_of_substance = uom::si::amount_of_substance::mole, electric_current = starship_battery::units::electric_current::ampere, length = uom::si::length::meter, luminous_intensity = uom::si::luminous_intensity::candela, mass = uom::si::mass::kilogram, thermodynamic_temperature = starship_battery::units::thermodynamic_temperature::kelvin, time = starship_battery::units::time::second> + 'static), f32>: Deserialize<'_>` is not satisfied
    --> /Users/andy/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tauri-plugin-system-info-0.2.0/src/model/common.rs:316:13
     |
316  |     energy: Energy,
     |             ^^^^^^ the trait `Deserialize<'_>` is not implemented for `Quantity<dyn Dimension<I = Z0, J = Z0, Kind = ..., L = ..., M = ..., N = ..., T = ..., Th = ...>, ..., ...>`
     |
     = help: the following other types implement trait `Deserialize<'de>`:
               <bool as Deserialize<'de>>
               <char as Deserialize<'de>>
               <isize as Deserialize<'de>>
               <i8 as Deserialize<'de>>
               <i16 as Deserialize<'de>>
               <i32 as Deserialize<'de>>
               <i64 as Deserialize<'de>>
               <i128 as Deserialize<'de>>
             and 295 others
note: required by a bound in `next_element`
    --> /Users/andy/.cargo/registry/src/index.crates.io-6f17d22bba15001f/serde-1.0.203/src/de/mod.rs:1726:12
     |
1724 |     fn next_element<T>(&mut self) -> Result<Option<T>, Self::Error>
     |        ------------ required by a bound in this associated function
1725 |     where
1726 |         T: Deserialize<'de>,
     |            ^^^^^^^^^^^^^^^^ required by this bound in `SeqAccess::next_element`

error[E0277]: the trait bound `uom::si::Quantity<(dyn uom::si::Dimension<I = Z0, J = Z0, Kind = (dyn uom::Kind + 'static), L = PInt<UInt<UInt<UTerm, B1>, B0>>, M = PInt<UInt<UTerm, B1>>, N = Z0, T = NInt<UInt<UInt<UTerm, B1>, B0>>, Th = Z0> + 'static), (dyn uom::si::Units<f32, amount_of_substance = uom::si::amount_of_substance::mole, electric_current = starship_battery::units::electric_current::ampere, length = uom::si::length::meter, luminous_intensity = uom::si::luminous_intensity::candela, mass = uom::si::mass::kilogram, thermodynamic_temperature = starship_battery::units::thermodynamic_temperature::kelvin, time = starship_battery::units::time::second> + 'static), f32>: Deserialize<'_>` is not satisfied
    --> /Users/andy/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tauri-plugin-system-info-0.2.0/src/model/common.rs:317:18
     |
317  |     energy_full: Energy,
     |                  ^^^^^^ the trait `Deserialize<'_>` is not implemented for `Quantity<dyn Dimension<I = Z0, J = Z0, Kind = ..., L = ..., M = ..., N = ..., T = ..., Th = ...>, ..., ...>`
     |
     = help: the following other types implement trait `Deserialize<'de>`:
               <bool as Deserialize<'de>>
               <char as Deserialize<'de>>
               <isize as Deserialize<'de>>
               <i8 as Deserialize<'de>>
               <i16 as Deserialize<'de>>
               <i32 as Deserialize<'de>>
               <i64 as Deserialize<'de>>
               <i128 as Deserialize<'de>>
             and 295 others
note: required by a bound in `next_element`
    --> /Users/andy/.cargo/registry/src/index.crates.io-6f17d22bba15001f/serde-1.0.203/src/de/mod.rs:1726:12
     |
1724 |     fn next_element<T>(&mut self) -> Result<Option<T>, Self::Error>
     |        ------------ required by a bound in this associated function
1725 |     where
1726 |         T: Deserialize<'de>,
     |            ^^^^^^^^^^^^^^^^ required by this bound in `SeqAccess::next_element`

error[E0277]: the trait bound `uom::si::Quantity<(dyn uom::si::Dimension<I = Z0, J = Z0, Kind = (dyn uom::Kind + 'static), L = PInt<UInt<UInt<UTerm, B1>, B0>>, M = PInt<UInt<UTerm, B1>>, N = Z0, T = NInt<UInt<UInt<UTerm, B1>, B0>>, Th = Z0> + 'static), (dyn uom::si::Units<f32, amount_of_substance = uom::si::amount_of_substance::mole, electric_current = starship_battery::units::electric_current::ampere, length = uom::si::length::meter, luminous_intensity = uom::si::luminous_intensity::candela, mass = uom::si::mass::kilogram, thermodynamic_temperature = starship_battery::units::thermodynamic_temperature::kelvin, time = starship_battery::units::time::second> + 'static), f32>: Deserialize<'_>` is not satisfied
    --> /Users/andy/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tauri-plugin-system-info-0.2.0/src/model/common.rs:318:25
     |
318  |     energy_full_design: Energy,
     |                         ^^^^^^ the trait `Deserialize<'_>` is not implemented for `Quantity<dyn Dimension<I = Z0, J = Z0, Kind = ..., L = ..., M = ..., N = ..., T = ..., Th = ...>, ..., ...>`
     |
     = help: the following other types implement trait `Deserialize<'de>`:
               <bool as Deserialize<'de>>
               <char as Deserialize<'de>>
               <isize as Deserialize<'de>>
               <i8 as Deserialize<'de>>
               <i16 as Deserialize<'de>>
               <i32 as Deserialize<'de>>
               <i64 as Deserialize<'de>>
               <i128 as Deserialize<'de>>
             and 295 others
note: required by a bound in `next_element`
    --> /Users/andy/.cargo/registry/src/index.crates.io-6f17d22bba15001f/serde-1.0.203/src/de/mod.rs:1726:12
     |
1724 |     fn next_element<T>(&mut self) -> Result<Option<T>, Self::Error>
     |        ------------ required by a bound in this associated function
1725 |     where
1726 |         T: Deserialize<'de>,
     |            ^^^^^^^^^^^^^^^^ required by this bound in `SeqAccess::next_element`

error[E0277]: the trait bound `uom::si::Quantity<(dyn uom::si::Dimension<I = Z0, J = Z0, Kind = (dyn uom::Kind + 'static), L = PInt<UInt<UInt<UTerm, B1>, B0>>, M = PInt<UInt<UTerm, B1>>, N = Z0, T = NInt<UInt<UInt<UTerm, B1>, B1>>, Th = Z0> + 'static), (dyn uom::si::Units<f32, amount_of_substance = uom::si::amount_of_substance::mole, electric_current = starship_battery::units::electric_current::ampere, length = uom::si::length::meter, luminous_intensity = uom::si::luminous_intensity::candela, mass = uom::si::mass::kilogram, thermodynamic_temperature = starship_battery::units::thermodynamic_temperature::kelvin, time = starship_battery::units::time::second> + 'static), f32>: Deserialize<'_>` is not satisfied
    --> /Users/andy/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tauri-plugin-system-info-0.2.0/src/model/common.rs:319:18
     |
319  |     energy_rate: Power,
     |                  ^^^^^ the trait `Deserialize<'_>` is not implemented for `Quantity<dyn Dimension<I = Z0, J = Z0, Kind = ..., L = ..., M = ..., N = ..., T = ..., Th = ...>, ..., ...>`
     |
     = help: the following other types implement trait `Deserialize<'de>`:
               <bool as Deserialize<'de>>
               <char as Deserialize<'de>>
               <isize as Deserialize<'de>>
               <i8 as Deserialize<'de>>
               <i16 as Deserialize<'de>>
               <i32 as Deserialize<'de>>
               <i64 as Deserialize<'de>>
               <i128 as Deserialize<'de>>
             and 295 others
note: required by a bound in `next_element`
    --> /Users/andy/.cargo/registry/src/index.crates.io-6f17d22bba15001f/serde-1.0.203/src/de/mod.rs:1726:12
     |
1724 |     fn next_element<T>(&mut self) -> Result<Option<T>, Self::Error>
     |        ------------ required by a bound in this associated function
1725 |     where
1726 |         T: Deserialize<'de>,
     |            ^^^^^^^^^^^^^^^^ required by this bound in `SeqAccess::next_element`

error[E0277]: the trait bound `uom::si::Quantity<(dyn uom::si::Dimension<I = NInt<UInt<UTerm, B1>>, J = Z0, Kind = (dyn uom::Kind + 'static), L = PInt<UInt<UInt<UTerm, B1>, B0>>, M = PInt<UInt<UTerm, B1>>, N = Z0, T = NInt<UInt<UInt<UTerm, B1>, B1>>, Th = Z0> + 'static), (dyn uom::si::Units<f32, amount_of_substance = uom::si::amount_of_substance::mole, electric_current = starship_battery::units::electric_current::ampere, length = uom::si::length::meter, luminous_intensity = uom::si::luminous_intensity::candela, mass = uom::si::mass::kilogram, thermodynamic_temperature = starship_battery::units::thermodynamic_temperature::kelvin, time = starship_battery::units::time::second> + 'static), f32>: Deserialize<'_>` is not satisfied
    --> /Users/andy/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tauri-plugin-system-info-0.2.0/src/model/common.rs:320:14
     |
320  |     voltage: ElectricPotential,
     |              ^^^^^^^^^^^^^^^^^ the trait `Deserialize<'_>` is not implemented for `Quantity<dyn Dimension<I = ..., J = ..., Kind = ..., L = ..., M = ..., N = ..., T = ..., Th = ...>, ..., ...>`
     |
     = help: the following other types implement trait `Deserialize<'de>`:
               <bool as Deserialize<'de>>
               <char as Deserialize<'de>>
               <isize as Deserialize<'de>>
               <i8 as Deserialize<'de>>
               <i16 as Deserialize<'de>>
               <i32 as Deserialize<'de>>
               <i64 as Deserialize<'de>>
               <i128 as Deserialize<'de>>
             and 295 others
note: required by a bound in `next_element`
    --> /Users/andy/.cargo/registry/src/index.crates.io-6f17d22bba15001f/serde-1.0.203/src/de/mod.rs:1726:12
     |
1724 |     fn next_element<T>(&mut self) -> Result<Option<T>, Self::Error>
     |        ------------ required by a bound in this associated function
1725 |     where
1726 |         T: Deserialize<'de>,
     |            ^^^^^^^^^^^^^^^^ required by this bound in `SeqAccess::next_element`

error[E0277]: the trait bound `uom::si::Quantity<(dyn uom::si::Dimension<I = Z0, J = Z0, Kind = (dyn uom::Kind + 'static), L = Z0, M = Z0, N = Z0, T = Z0, Th = Z0> + 'static), (dyn uom::si::Units<f32, amount_of_substance = uom::si::amount_of_substance::mole, electric_current = starship_battery::units::electric_current::ampere, length = uom::si::length::meter, luminous_intensity = uom::si::luminous_intensity::candela, mass = uom::si::mass::kilogram, thermodynamic_temperature = starship_battery::units::thermodynamic_temperature::kelvin, time = starship_battery::units::time::second> + 'static), f32>: Deserialize<'_>` is not satisfied
    --> /Users/andy/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tauri-plugin-system-info-0.2.0/src/model/common.rs:321:22
     |
321  |     state_of_health: Ratio,
     |                      ^^^^^ the trait `Deserialize<'_>` is not implemented for `Quantity<dyn Dimension<I = Z0, J = Z0, Kind = dyn Kind, L = Z0, M = Z0, N = Z0, T = Z0, Th = Z0>, ..., ...>`
     |
     = help: the following other types implement trait `Deserialize<'de>`:
               <bool as Deserialize<'de>>
               <char as Deserialize<'de>>
               <isize as Deserialize<'de>>
               <i8 as Deserialize<'de>>
               <i16 as Deserialize<'de>>
               <i32 as Deserialize<'de>>
               <i64 as Deserialize<'de>>
               <i128 as Deserialize<'de>>
             and 295 others
note: required by a bound in `next_element`
    --> /Users/andy/.cargo/registry/src/index.crates.io-6f17d22bba15001f/serde-1.0.203/src/de/mod.rs:1726:12
     |
1724 |     fn next_element<T>(&mut self) -> Result<Option<T>, Self::Error>
     |        ------------ required by a bound in this associated function
1725 |     where
1726 |         T: Deserialize<'de>,
     |            ^^^^^^^^^^^^^^^^ required by this bound in `SeqAccess::next_element`

error[E0277]: the trait bound `uom::si::Quantity<(dyn uom::si::Dimension<I = Z0, J = Z0, Kind = (dyn uom::si::marker::TemperatureKind + 'static), L = Z0, M = Z0, N = Z0, T = Z0, Th = PInt<UInt<UTerm, B1>>> + 'static), (dyn uom::si::Units<f32, amount_of_substance = uom::si::amount_of_substance::mole, electric_current = starship_battery::units::electric_current::ampere, length = uom::si::length::meter, luminous_intensity = uom::si::luminous_intensity::candela, mass = uom::si::mass::kilogram, thermodynamic_temperature = starship_battery::units::thermodynamic_temperature::kelvin, time = starship_battery::units::time::second> + 'static), f32>: Deserialize<'_>` is not satisfied
    --> /Users/andy/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tauri-plugin-system-info-0.2.0/src/model/common.rs:324:24
     |
324  |     temperature_kelin: Option<ThermodynamicTemperature>,
     |                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Deserialize<'_>` is not implemented for `Quantity<dyn Dimension<I = Z0, J = Z0, Kind = ..., L = ..., M = ..., N = ..., T = ..., Th = ...>, ..., ...>`, which is required by `std::option::Option<uom::si::Quantity<(dyn uom::si::Dimension<I = Z0, J = Z0, Kind = (dyn uom::si::marker::TemperatureKind + 'static), L = Z0, M = Z0, N = Z0, T = Z0, Th = PInt<UInt<UTerm, B1>>> + 'static), (dyn uom::si::Units<f32, amount_of_substance = uom::si::amount_of_substance::mole, electric_current = starship_battery::units::electric_current::ampere, length = uom::si::length::meter, luminous_intensity = uom::si::luminous_intensity::candela, mass = uom::si::mass::kilogram, thermodynamic_temperature = starship_battery::units::thermodynamic_temperature::kelvin, time = starship_battery::units::time::second> + 'static), f32>>: Deserialize<'_>`
     |
     = help: the following other types implement trait `Deserialize<'de>`:
               <bool as Deserialize<'de>>
               <char as Deserialize<'de>>
               <isize as Deserialize<'de>>
               <i8 as Deserialize<'de>>
               <i16 as Deserialize<'de>>
               <i32 as Deserialize<'de>>
               <i64 as Deserialize<'de>>
               <i128 as Deserialize<'de>>
             and 295 others
     = note: required for `Option<Quantity<..., ..., ...>>` to implement `Deserialize<'_>`
note: required by a bound in `next_element`
    --> /Users/andy/.cargo/registry/src/index.crates.io-6f17d22bba15001f/serde-1.0.203/src/de/mod.rs:1726:12
     |
1724 |     fn next_element<T>(&mut self) -> Result<Option<T>, Self::Error>
     |        ------------ required by a bound in this associated function
1725 |     where
1726 |         T: Deserialize<'de>,
     |            ^^^^^^^^^^^^^^^^ required by this bound in `SeqAccess::next_element`
     = note: the full name for the type has been written to '/Users/andy/Development/Tauri/photo-importer/src-tauri/target/debug/deps/tauri_plugin_system_info-f921e2b9bc7ad894.long-type-1585738714020904999.txt'
     = note: consider using `--verbose` to print the full type name to the console
```
</details>

**Comment by HuakunShen** [comment]
What version of Tauri are you using?

**Comment by atwright147** [comment]
Sorry, was tired last night. I think it's `1.6.7`

To recreate, just create a new Tauri app. This is what I did:

1. `npx create-tauri-app`
1. `cd {project_dir}`
1. `npm i`
1. `cd src-tauri`
1. `cargo add tauri-plugin-system-info`
1. `cd ..`
1. `npm run tauri dev`

Tauri version based on this section from the Cargo lock file

```cargo
[[package]]
name = "tauri"
version = "1.6.7"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "67c7177b6be45bbb875aa239578f5adc982a1b3d5ea5b315ccd100aeb0043374"
dependencies = [
 "anyhow",
 "cocoa",
 "dirs-next",
 "dunce",
 "embed_plist",
 "encoding_rs",
 "flate2",
 "futures-util",
 "getrandom 0.2.15",
 "glib",
 "glob",
 "gtk",
 "heck 0.5.0",
 "http",
 "ignore",
 "objc",
 "once_cell",
 "open",
 "percent-encoding",
 "rand 0.8.5",
 "raw-window-handle",
 "regex",
 "semver",
 "serde",
 "serde_json",
 "serde_repr",
 "serialize-to-javascript",
 "state",
 "tar",
 "tauri-macros",
 "tauri-runtime",
 "tauri-runtime-wry",
 "tauri-utils",
 "tempfile",
 "thiserror",
 "tokio",
 "url",
 "uuid",
 "webkit2gtk",
 "webview2-com",
 "windows 0.39.0",
]
```

NB. I am just starting out with Tauri and Rust, so don't discount how stupid I probably am :D

**Comment by atwright147** [comment]
Any thoughts? Can you recreate the issue?

**Comment by HuakunShen** [comment]
Yeah,there is some serialization error.
Not sure why. It used to work. No dependengcy version has changed. I am working on fixing it.

**Comment by HuakunShen** [comment]
Should now be fixed in v1.0.0

**Comment by atwright147** [comment]
Wahoo! Yes, that fixed it

Thank you very much
